src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -1,33 +1,154 @@
 package seedu.address.logic.parser;
 
-import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
-
-import java.util.Arrays;
-
 import seedu.address.logic.commands.FindCommand;
 import seedu.address.logic.parser.exceptions.ParseException;
 import seedu.address.model.person.NameContainsKeywordsPredicate;
 
+import java.util.ArrayList;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.model.person.Module.MESSAGE_CONSTRAINTS;
+import static seedu.address.model.person.Module.VALIDATION_REGEX;
+
 /**
  * Parses input arguments and creates a new FindCommand object
  */
 public class FindCommandParser implements Parser<FindCommand> {
 
+    // Serves as regex to match for name prefix
+    private final Prefix namePrefix = new Prefix("n/");
+
+    // Serves as regex to match for module prefix
+    private final Prefix modulePrefix = new Prefix("m/");
+
+    // Serves as regex to match for any whitespace with an OR
+    private final String orWhitespace = "|\\s+";
+
+    // Serves as regex to match for any whitespace
+    private final String whitespace = "\\s+";
+
+    // Serves as regex to match for both name and module prefix, along with whitespaces
+    private final Prefix nameModulePrefix = new Prefix("(\\s*([nm]/))+|(\\s+)");
+
     /**
      * Parses the given {@code String} of arguments in the context of the FindCommand
      * and returns a FindCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
         String trimmedArgs = args.trim();
+
         if (trimmedArgs.isEmpty()) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
+        // Regex depends on user input
+        String regex;
+
+        Pattern namePattern = Pattern.compile(namePrefix.getPrefix());
+        Pattern modulePattern = Pattern.compile(modulePrefix.getPrefix());
+        Matcher nameMatcher = namePattern.matcher(trimmedArgs);
+        Matcher moduleMatcher = modulePattern.matcher(trimmedArgs);
+        boolean hasName = nameMatcher.find();
+        boolean hasModule = moduleMatcher.find();
+
+        // Represents simply which branch it matches
+        int mode;
```

Comment:
maybe for the mode value use enum? Eg. 0 is NAME_AND_MODULE, 1 IS NAME_ONLY, 2 IS MODULE_ONLY

Reply:
Great suggestion, I'll add that in.